### PR TITLE
Adding context to errors around network failures

### PR DESCRIPTION
**Pull request by nhibberd**

### CODE REVIEW

**Comment by markhibberd** [comment]
:+1: Although somewhat superficial because I don't know the code that well. The extra error messages seem legit. Might be good to get another set of eyes and a :green_apple: before merging though.

File: src/Ambiata/Cli/Processing.hs [hunk 1]
```diff
@@ -32,14 +26,18 @@ import qualified Data.Text                    as T
...
+import           System.FilePath              hiding ((</>))
+import           System.IO
 
 -- |
 -- Upload the files that are in the processing dir, and move to archive when done.
 --
 uploadReady :: IncomingDir -> Region -> UploadAccess -> EitherT AmbiataError IO [ArchivedFile]
 uploadReady dir r (UploadAccess (TemporaryAccess (TemporaryCreds k s sess) a)) =
-  firstEitherT AmbiataAWSError $ runAWSWithCreds r k s (Just $ sess) $ processReady dir a
+  bimapEitherT AmbiataAWSError id $ runAWSWithCreds r k s (Just $ sess) $ processReady dir a
```
Comment (charleso) [inline src/Ambiata/Cli/Processing.hs]: Should this have more context (eg. upload?)

File: src/Ambiata/Cli/Credentials.hs [hunk 2]
```diff
@@ -48,7 +49,7 @@ obtainCredentials :: PermissionDirection -> AmbiataAPIKey -> AmbiataAPIEndpoint
 obtainCredentials pd key' (AmbiataAPIEndpoint ep)  = do
   m <- liftIO $ newManager tlsManagerSettings
   req <- parseUrl $ T.unpack ep
-  res <- liftIO . httpGo m $ getTokenRequest pd req key'
+  res <- (liftIO . httpGo m $ getTokenRequest pd req key') `catch` (\(e :: HttpException) -> left $ NetworkException e)
```
Comment (charleso) [inline src/Ambiata/Cli/Credentials.hs]: Should we add context here too?

**Comment by charleso** [comment]
:+1: Once :green_apple: and any appropriate context we can add

**Comment by nhibberd** [comment]
Bump

File: src/Ambiata/Cli/Data.hs [hunk 3]
```diff
@@ -150,59 +151,83 @@ data RunMode = Daemon | OneShot
...
-renderCredentialError (BadResponse t) = "An error occurred with the Ambiata API: " <> t
-renderCredentialError (DecodeError t) = "Incorrect JSON returned from Ambiata API: " <> t
+renderCredentialError e =
+  case e of
+    BadResponse t ->
+      "An error occurred with the Ambiata API: " <> t
+    DecodeError t ->
+      "Incorrect JSON returned from Ambiata API: " <> t
+    NetworkException t ->
+      "A network error has occured with the Ambiata API: " <> (T.pack $ show t)
```
Comment (olorin) [inline src/Ambiata/Cli/Data.hs]: <nitpick>I'm not sure I'd call an HTTP exception a network error; in most cases it's a protocol or application error (with a few exceptions like `ResponseTimeout`). Doesn't really matter though.</nitpick>

**Comment by olorin** [comment]
:8ball:

**Pull request merged**